BazhenovYN/rss-react-app
# React. Tests. Context API

1. Task: [link](https://github.com/rolling-scopes-school/tasks/tree/master/react/modules/module03)
2. Screenshot:
![image](https://github.com/BazhenovYN/rss-react-app/assets/114768651/9abd5522-b3e3-4e25-91ce-dda8749ce7ba)
3. Deploy: [link](https://rss-sw.netlify.app/)
4. Done 12.11.2023 / deadline 12.11.2023
5. Score: 100 / 100
  - [x] Custom state management using the Context API is implemented - 5 points
  - [x] The Search component value is stored in the context - 5 points
  - [x] The list of items received from the API is stored in the context - 5 points
  - [x] The context is used in components that need access to the data - 5 points
  - [x] React Testing Library and Jest or Vitest are added and configured - 10 points
  - [x] Husky runs tests on pre-push - 10 points
  - [x] Test cases - 60 points (5 points per each)

1) Tests for the Card List component:
- Verify that the component renders the specified number of cards `(CardList.test.tsx, test1)`;
- Check that an appropriate message is displayed if no cards are present `(CardList.test.tsx, test2)`.

2) Tests for the Card component:
- Ensure that the card component renders the relevant card data `(Card.test.tsx, test1)`;
- Validate that clicking on a card opens a detailed card component `(Card.test.tsx, test2)`;
- Check that clicking triggers an additional API call to fetch detailed information `(Card.test.tsx, test2)`.

3) Tests for the Detailed Card component:
- Check that a loading indicator is displayed while fetching data `(DetailCard.test.tsx, test2)`;
- Make sure the detailed card component correctly displays the detailed card data `(DetailCard.test.tsx, test1)`;
- Ensure that clicking the close button hides the component `(DetailCard.test.tsx, test3)`.

4) Tests for the Pagination component:
- Make sure the component updates URL query parameter when page changes `(Pagination.test.tsx, test4)`.

5) Tests for the Search component:
- Verify that clicking the Search button saves the entered value to the local storage `(SearchView.test.tsx, test2)`;
- Check that the component retrieves the value from the local storage upon mounting `(SearchView.test.tsx, test3)`.

6) Tests for the 404 Page component:
- Ensure that the 404 page is displayed when navigating to an invalid route `(NotFound.test.tsx, test2)`.
